fetch-in-react-practice/client/src/components/Todos.tsx

```diff
@@ -0,0 +1,156 @@
+import { useEffect, useState } from 'react';
+import { PageTitle } from './PageTitle';
+import { TodoList } from './TodoList';
+import { TodoForm } from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function getTodos() {
+      try {
+        const res = await fetch('/api/todos');
+        if (!res.ok) throw new Error(`${res.status}`);
+        const list = await res.json();
+        setTodos(list);
+      } catch (err) {
+        setError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    getTodos();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo: UnsavedTodo) {
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(newTodo),
+    };
+    try {
+      const res = await fetch('/api/todos', options);
+      if (!res.ok) throw new Error(`${res.status}`);
+      const todo = await res.json();
+      const newTodos = todos.concat(todo);
+      setTodos(newTodos);
+    } catch (err) {
+      setError(err);
+    } finally {
+      setIsLoading(false);
```

Comment:
isLoading will already be false (or you couldn't have gotten here in the code). I don't think this hurts, but it's not necessary.